services/gundeck/src/Gundeck/Push/Websocket.hs

```diff
@@ -99,7 +99,8 @@ logBadCannons (uri, (err, prcs)) = do
     forM_ prcs $ \prc ->
         Log.warn $ logPresence prc
             ~~ Log.field "created_at" (ms $ createdAt prc)
-            ~~ Log.msg (val "WebSocket presence unreachable: " +++ (toByteString $ show (uri, err, resource prc)))
+            ~~ Log.msg (val "WebSocket presence unreachable: " +++
+                        show (uri, resource prc, show err))
```

Comment:
Perhaps remove the duplicate `show`?

Reply:
Also it might be better to report the URI and resource via `Log.field`.

Reply:
Oh wait, the duplicate `show` is actually what makes things work. There should be a comment about it, then.

Reply:
Ok, I will elaborate on this in a different PR.